dkpro-core-io-bincas-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/bincas/BinaryCasReader.java

```diff
@@ -76,54 +102,95 @@ public void getNext(CAS aCAS)
             // Prepare for format detection
             bis.mark(32);
             DataInputStream dis = new DataInputStream(bis);
-            byte[] header = new byte[DKPRO_HEADER.length];
             dis.read(header);
             
-            // Use externalized type system
-            TypeSystemImpl ts = null;
-            if (typeSystemLocation != null) {
-                // Otherwise try reading an externalized type system
-                ts = readTypeSystem();
-                
-                // If we encounter a Java-serialized file with an external TSI, then we
-                // reinitalize the CAS with the external TSI prior to loading the data
-                if (header[0] == (byte) 0xAC && header[1] == (byte) 0xED) {
-                    CASMgrSerializer casMgr = readCasManager();
-                    ((CASImpl)aCAS).setupCasFromCasMgrSerializer(casMgr);
-                }
-            }
-            
-            // BEGIN -- Legacy DKPro file format support
-            // Check if this is original UIMA CAS format or DKPro Core format
+ 			// If it is DKPro Core format, read the type system
             if (Arrays.equals(header, DKPRO_HEADER)) {
-                // If it is DKPro Core format, read the type system
-                getLogger().debug("Found DKPro-Core-style embedded type system");
-                ObjectInputStream ois = new ObjectInputStream(bis);
-                CASMgrSerializer casMgr = (CASMgrSerializer) ois.readObject();
-                if (ts == null) {
-                    ts = casMgr.getTypeSystem();
-                    ts.commit();
-                }
-            }
-            else {
-                // No embedded TS, reset 
-                bis.reset();
-            }
-            // END -- Legacy DKPro file format support
+				xts = readDKProHeader(bis, header, xts);
+	 		} else {
+	 			// No embedded DKPro TS, reset
+	 			try {
+	 				bis.reset();
+	 			} catch (IOException e) {
+	 				throw new IOException(e);
```

Comment:
Catching an exception and then throwing the same type of exception is redundant. Just don't try/catch.